cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+
+from cfme import test_requirements
+from cfme.base.credential import Credential
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.utils.generators import random_vm_name
+
+from widgetastic.utils import partial_match
+
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.long_running,
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([AzureProvider], scope='function',
+                         required_fields=[["provisioning", "image"]]),
+
+]
+
+
+def new_credential():
+    return Credential(principal='uid{}'.format(fauxfactory.gen_alphanumeric(4)),
+                      secret='redhat')
```

Comment:
This should be auto-generated as well so that we don't have simple passwords on public cloud entities.